File: Java/OS/DatabaseConnection.java
package OS;

import java.sql.*;

public class DatabaseConnection {

    public static Connection get(String name,String passwd) {
        Connection conn=null;
        String jdbc_driver="com.mysql.cj.jdbc.Driver";
        String jdbc_url="jdbc:mysql://localhost:3306;DatabaseName=OS";
        try{
            Class.forName(jdbc_driver);
            conn=DriverManager.getConnection(jdbc_url,"name","passwd");
        }catch(ClassNotFoundException|SQLException ex){
            ex.printStackTrace();
        }
        return conn;
    }
    public static void close(ResultSet rs,PreparedStatement pstmt,Connection conn) throws Exception {
        if (rs != null) {
            rs.close();
        }
        if (pstmt!= null) {
            pstmt.close();
        }
        if (conn != null) {
            conn.close();
        }
    }
    public static void close(PreparedStatement pstmt,Connection conn) throws Exception {
        if (pstmt!= null) {
            pstmt.close();
        }
        if (conn != null) {
            conn.close();
        }
    }
}

File: Java/StuInfoManageSystem/src/main/kotlin/sims/DatabaseConnection.kt
package sims

import java.sql.*

object DatabaseConnection {
	@JvmStatic
	fun get(): Connection? {
		var conn: Connection? = null
		val jdbcUrl = "jdbc:oracle:thin:@localhost:1521:orcl"
		try {
			conn = DriverManager.getConnection(jdbcUrl, "system", "Yan011017")
		} catch (ex: ClassNotFoundException) {
			ex.printStackTrace()
		} catch (ex: SQLException) {
			ex.printStackTrace()
		}
		return conn
	}
	
	@Throws(Exception::class)
	fun close(rs: ResultSet?, statement: PreparedStatement?, conn: Connection?) {
		rs?.close()
		statement?.close()
		conn?.close()
	}
	
	@Throws(Exception::class)
	fun close(statement: PreparedStatement?, conn: Connection?) {
		statement?.close()
		conn?.close()
	}
	
	@JvmStatic
	@Throws(Exception::class)
	fun close(conn: Connection?) {
		conn?.close()
	}
}